### PR TITLE
Delete deprecated endpoints

### PR DESCRIPTION
**Pull request by mhamra**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/17781|

## Description

Delete endpoints marked with @deprecate_endpoint and related artifacts.

### CODE REVIEW

**Comment by mhamra** [comment]
### UPDATE

Running the tavern file "test_rbac_black_security_endpoints.tavern.yaml" in a local environment using pytest/docker-compose v2 passed all tests.

```
pytest -vv test_rbac_black_security_endpoints.tavern.yaml
=========================================================================================== test session starts 
===========================================================================================
platform linux -- Python 3.9.16, pytest-7.3.1, pluggy-1.0.0 -- /home/marcelo/wazuh/wazuh/venv-3.9/bin/python
cachedir: .pytest_cache
metadata: {'Python': '3.9.16', 'Platform': 'Linux-5.15.0-1041-azure-x86_64-with-glibc2.35', 'Packages': {'pytest': '7.3.1', 'pluggy': '1.0.0'}, 'Plugins': {'aiohttp': '1.0.4', 'html': '2.1.1', 'tavern': '1.23.5', 'asyncio': '0.18.1', 'metadata': '3.0.0', 'trio': '0.7.0'}}
rootdir: /home/marcelo/wazuh/wazuh/api/test/integration
configfile: pytest.ini
plugins: aiohttp-1.0.4, html-2.1.1, tavern-1.23.5, asyncio-0.18.1, metadata-3.0.0, trio-0.7.0
asyncio: mode=auto
collected 25 items                                                                                                                                                                                        

test_rbac_black_security_endpoints.tavern.yaml::GET USERS RBAC PASSED                                                                                                                               [  4%]
test_rbac_black_security_endpoints.tavern.yaml::GET ROLES RBAC PASSED                                                                                                                               [  8%]
test_rbac_black_security_endpoints.tavern.yaml::GET RULES RBAC PASSED                                                                                                                               [ 12%]
test_rbac_black_security_endpoints.tavern.yaml::GET POLICIES RBAC PASSED                                                                                                                            [ 16%]
test_rbac_black_security_endpoints.tavern.yaml::GET SECURITY CONFIG PASSED                                                                                                                          [ 20%]
test_rbac_black_security_endpoints.tavern.yaml::UPDATE SECURITY CONFIG PASSED                                                                                                                       [ 24%]
test_rbac_black_security_endpoints.tavern.yaml::UPDATE USERS RBAC PASSED                                                                                                                            [ 28%]
test_rbac_black_security_endpoints.tavern.yaml::UPDATE USER'S ALLOW_RUN_AS FLAG PASSED                                                                                                              [ 32%]
test_rbac_black_security_endpoints.tavern.yaml::UPDATE ROLES RBAC PASSED                                                                                                                            [ 36%]
test_rbac_black_security_endpoints.tavern.yaml::UPDATE RULES RBAC PASSED                                                                                                                            [ 40%]
test_rbac_black_security_endpoints.tavern.yaml::UPDATE POLICIES RBAC PASSED                                                                                                                         [ 44%]
test_rbac_black_security_endpoints.tavern.yaml::CREATE LINK USER-ROLES RBAC PASSED                                                                                                                  [ 48%]
test_rbac_black_security_endpoints.tavern.yaml::CREATE USERS RBAC PASSED                                                                                                                            [ 52%]
test_rbac_black_security_endpoints.tavern.yaml::CREATE ROLES AND POLICIES RBAC PASSED                                                                                                               [ 56%]
test_rbac_black_security_endpoints.tavern.yaml::DELETE LINK ROLE-POLICIES RBAC PASSED                                                                                                               [ 60%]
test_rbac_black_security_endpoints.tavern.yaml::CREATE LINK ROLE-POLICIES RBAC PASSED                                                                                                               [ 64%]
test_rbac_black_security_endpoints.tavern.yaml::DELETE LINK ROLE-POLICIES RBAC PASSED                                                                                                               [ 64%]
test_rbac_black_security_endpoints.tavern.yaml::CREATE LINK ROLE-RULES RBAC PASSED                                                                                                                  [ 68%]
test_rbac_black_security_endpoints.tavern.yaml::DELETE LINK ROLE-RULES RBAC PASSED                                                                                                                  [ 72%]
test_rbac_black_security_endpoints.tavern.yaml::DELETE USERS RBAC PASSED                                                                                                                            [ 76%]
test_rbac_black_security_endpoints.tavern.yaml::DELETE RULES RBAC PASSED                                                                                                                            [ 80%]
test_rbac_black_security_endpoints.tavern.yaml::DELETE /security/rules (invalid) PASSED                                                                                                             [ 84%]
test_rbac_black_security_endpoints.tavern.yaml::DELETE POLICIES RBAC PASSED                                                                                                                         [ 88%]
test_rbac_black_security_endpoints.tavern.yaml::DELETE ROLES RBAC PASSED                                                                                                                            [ 92%]
test_rbac_black_security_endpoints.tavern.yaml::REVOKE TOKENS RBAC PASSED   

============================================================================================ warnings summary =============================================================================================
../../../venv-3.9/lib/python3.9/site-packages/pytest_aiohttp/plugin.py:28
  /home/marcelo/wazuh/wazuh/venv-3.9/lib/python3.9/site-packages/pytest_aiohttp/plugin.py:28: DeprecationWarning: The 'asyncio_mode' is 'legacy', switching to 'auto' for the sake of pytest-aiohttp backward compatibility. Please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto' in pytest configuration file.
    config.issue_config_time_warning(LEGACY_MODE, stacklevel=2)

../../../venv-3.9/lib/python3.9/site-packages/_pytest/nodes.py:642
  /home/marcelo/wazuh/wazuh/venv-3.9/lib/python3.9/site-packages/_pytest/nodes.py:642: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to YamlFile is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return super().from_parent(parent=parent, fspath=fspath, path=path, **kw)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================================================== 25 passed, 2 warnings in 839.73s (0:13:59) ================================================================================

```

**Comment by mhamra** [comment]
### UPDATE
- Branch rebased to master
- runnin test_black_rbac_security_endpointes.tavern.yaml in the development environment doesn't fail.

File: api/api/spec/spec.yaml [hunk 1]
```diff
@@ -10529,7 +10468,7 @@ paths:
       description: "Return Wazuh statistical information in node {node_id} for the current or specified date"
       operationId: api.controllers.cluster_controller.get_stats_node
       x-rbac-actions:
-        - $ref: '#/x-rbac-catalog/actions/cluster:read'
+        - $ref: '#/x-rbac-catalog/actions/cluster:read'
```
Comment (GGP1) [inline api/api/spec/spec.yaml:10471]:
```suggestion
        - $ref: '#/x-rbac-catalog/actions/cluster:read'
```

**Review by GGP1 — CHANGES_REQUESTED** [review]
We should remove the endpoints from the [QA API integrations tests](https://github.com/wazuh/wazuh-qa/tree/master/tests/integration/test_api) as well. Some tests are failing here https://github.com/wazuh/wazuh/pull/17815/checks?check_run_id=15491012104 because they call endpoints that were removed.

**Review by GGP1 — CHANGES_REQUESTED** [review]
@mhamra I don't see any changes since my last review.

The API IT tier 0,1 [check](https://github.com/wazuh/wazuh/pull/17815/checks?check_run_id=15491012104) is failing because the deleted endpoints still exist in those tests. We should create an issue in the [wazuh-qa](https://github.com/wazuh/wazuh-qa) repository to fix them. Once that issue is finished, we can run the check again and continue with this issue's workflow.

File: framework/wazuh/agent.py [hunk 2]
```diff
@@ -1312,10 +1312,7 @@ def get_agent_config(agent_list: list = None, component: str = None, config: str
                   post_proc_kwargs={'exclude_codes': [1701, 1703]})
 def get_agents_sync_group(agent_list: list = None) -> AffectedItemsWazuhResult:
     """Get agents configuration sync status.
-
-    Notes
-    -----
-    To be deprecated in v5.0.
+    Used by script agent_groups --show-sync option
```
[the remark this answers is not in the document]
Comment (mhamra) [reply framework/wazuh/agent.py:1315]: The get_agents_sync_group calls the get_agents_info function to query the database. The function get_agents_info is not deprecated and is used in many places throughout the framework code. The script shows if a specified agent has its group configuration synchronized and is still useful. We better maintain this option in the script/command.

**Comment by mhamra** [comment]
### UPDATE

Integration API Test Tier "0", "1".

The API integration tests are currently running in Jenkins and Github actions. Jenkins tests will soon be removed, and the only valid tests will run in Github. [This check](https://github.com/wazuh/wazuh/actions/runs/5832939979/job/15819301642?pr=17815) shows the results of the execution.

Searching in the source code of the tests, I couldn't find calls to the endpoints that this PR will remove.

**Review by GGP1 — APPROVED** [review]
LGTM!

> The failed check `Test: API IT - Tier 0,1 -manager ` was replaced with `Integration tests for API - Tier 0 and 1` and will be removed soon

**Comment by Selutario** [comment]
This was blocked for a long time since there was no dev branch for 5.0.0, and now there are too many conflicts to fix. Since many services will be removed in 5.0.0:
- https://github.com/wazuh/wazuh/issues/22888

Lots of API endpoints will need to be deleted too, as listed in https://github.com/wazuh/wazuh/issues/23393#issuecomment-2163818486. Therefore, I'm closing this PR and issue and we will re-evaluate this as part of https://github.com/wazuh/wazuh/issues/22677.

**Pull request closed without merging**